samples/ReadMe/Provider.Tests/SomethingApiFixture.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace ReadMe.Provider.Tests
+{
+    public class SomethingApiFixture : IDisposable
+    {
+        private readonly IHost server;
+        public Uri ServerUri { get; }
+
+        public SomethingApiFixture()
+        {
+            ServerUri = new Uri("http://localhost:9222");
+            server = Host.CreateDefaultBuilder()
+                        .ConfigureWebHostDefaults(webBuilder =>
+                        {
+                            webBuilder.UseUrls(ServerUri.ToString());
+                            webBuilder.UseStartup<TestStartup>();
+                        })
+                        .Build();
+            server.Start();
+        }
+
+        public void Dispose()
+        {
+            server.Dispose();
+
+            // Clean out leftover data
+            foreach (var dataFile in Directory.GetFiles(Path.Combine("..", "..", "..", "data")))
+            {
+                File.Delete(dataFile);
```

Comment:
This is probably going to make the tests annoying to run because you'll always need to remember to run the consumer test first and then the provider test later to delete the file (which makes CI a bit awkward given they're sharing state effectively).

The other samples check the pact file in and never delete it.